test/e2e/tests/test_hpo.py

```diff
@@ -61,26 +76,102 @@ def get_sagemaker_hpo_job(sagemaker_client, hpo_job_name):
+    list_status_created = ("InProgress", "Completed")
+    list_status_stopped = ("Stopped", "Stopping")
+    status_inprogress: str = "InProgress"
+    status_completed: str = "Completed"
+
+    def _wait_resource_hpo_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+        expected_status, wait_periods, period_length, get_hpo_resource_status, reference
+    )
+
+    def _wait_sagemaker_hpo_status(
+        self, 
+        hpo_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_hpo_sagemaker_status,
+            hpo_job_name,
+        )
+
+    def _assert_hpo_status_in_sync(
+        self, hpo_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_hpo_status(
+                hpo_job_name, expected_status
+            )
+            == self._wait_resource_hpo_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_hpo(self, xgboost_hpojob):
+        (reference, resource) = xgboost_hpojob
+        assert k8s.get_resource_exists(reference)
+    
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+        assert hpo_job_name is not None
+
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+        assert k8s.get_resource_arn(resource) == hpo_sm_desc["HyperParameterTuningJobArn"]
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_created
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        self._assert_hpo_status_in_sync(
+            hpo_job_name, reference, self.status_inprogress
+        )
+
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted is True
+
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_stopped
+
+    def test_completed_hpo(self, xgboost_hpojob):
         (reference, resource) = xgboost_hpojob
         assert k8s.get_resource_exists(reference)
 
         hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
         assert hpo_job_name is not None
 
-        hpo_sm_desc = get_sagemaker_hpo_job(sagemaker_client, hpo_job_name)
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
         assert (
             k8s.get_resource_arn(resource) == hpo_sm_desc["HyperParameterTuningJobArn"]
         )
-        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in HPO_JOB_STATUS_CREATED
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_created
```

Comment:
_assert_hpo_status_in_sync ?
nit: remove _ prefix from method names

Reply:
I am following the naming across all files here, or did i miss something ?

Reply:
ya, this was comments in one of the previous PRs, doesn't make sense to prefix with _
ok for now, not a blocker